cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -96,7 +97,8 @@ constexpr bool is_supported_encoding(Encoding enc)
 /**
  * @brief Atomically OR `error` into `error_code`.
  */
-constexpr void set_error(kernel_error::value_type error, kernel_error::pointer error_code)
+__device__ constexpr void set_error(kernel_error::value_type error,
+                                    kernel_error::pointer error_code)
```

Comment:
ideally we should move all device and host-device functions to a CUDA header

Reply:
should we make these headers `.cuh`?

Reply:
Ideally, yes. @vyasr and I discussed this, and we agreed that the potential risk is that many function files need to be migrated to `cuh` headers, and C++ headers will need to be renamed as CUDA headers. This could introduce breaking changes, especially for public APIs. The current focus is on making libcudf work without relying on relaxed constexpr. Follow-up cleanup tasks, such as moving host-device code to CUDA files, are desired and should be addressed separately, as they need to be carefully evaluated to ensure they won't cause significant issues for libcudf users.